source/extensions/filters/listener/tls_inspector/tls_inspector.cc

```diff
@@ -130,6 +130,81 @@ void Filter::onServername(absl::string_view name) {
   clienthello_success_ = true;
 }
 
+void Filter::extractEcdsaCipherSuites(const SSL_CLIENT_HELLO* ctx) {
```

Comment:
Nit: could you rename `ctx` to something more appropriate? `SSL_CLIENT_HELLO` used to be called `ssl_early_callback_ctx`, which explains the variable name that you might find internally.

Reply:
Also, this function is missing support for `signature_algorithms`.

Reply:
I couldn't tell from https://tools.ietf.org/html/rfc5246#section-7.4.1.4.1 if we have to support this, or whether we can rely on the cipher suite/curve heuristics in this method. All things being equal, I would like to keep this as simple as possible while still being capable of doing the job until we subsume this functionality into BoringSSL.